src/xdocs/checks/sizes/parameternumber.xml

```diff
@@ -112,6 +119,60 @@
 public void needsLotsOfParameters(int a,
     int b, int c, int d, int e, int f, int g, int h) {
     ...
+}
+        </source>
+        <p id="Example4-config">
+          To configure the check to ignore methods and constructors annotated with
+          <code>JsonCreator</code> annotation:
+        </p>
+        <source>
+&lt;module name=&quot;Checker&quot;&gt;
+  &lt;module name=&quot;TreeWalker&quot;&gt;
+    &lt;module name=&quot;ParameterNumber&quot;&gt;
+      &lt;property name=&quot;ignoreAnnotatedBy&quot; value=&quot;JsonCreator&quot;/&gt;
+    &lt;/module&gt;
+  &lt;/module&gt;
+&lt;/module&gt;
+        </source>
+        <p id="Example4-code"> Example:</p>
+        <source>
+class Example {
+    @JsonCreator
+    Example(int a, int b, int c, int d,
+            int e, int f, int g, int h) {}
+
+    @interface JsonCreator {}
+}
+        </source>
+        <p>
+          Fully qualified class name(s) can also specified in ignoreAnnotatedBy property.
+          Note that when specifying annotation(s), only methods and constructors annotated with
+          the exact specified annotation(s) will be ignored, any variation, even if it refers to
+          the same annotation, will result in it being checked, and produce violation if any.
+        </p>
+        <p id="Example5-config">
```

Comment:
Please completely remove this example, for two reasons:
1. With my proposed example above, we accurately demonstrate the check's behavior in this respect
2. I can't think of any practical reason that anyone would want to ignore the number of parameters in a method definition based on the method is `Deprecated` or not. It is OK (even desireable) to use weird code in tests, but in user facing documentation, all examples should have reasonable usage that makes sense and is familiar to developers.

@sktpy what are your thoughts on this?

Reply:
Also, apologies if we had a misunderstanding at https://github.com/checkstyle/checkstyle/pull/14553#discussion_r1505368278, my intention was to show the check's behavior on some code like this, not to make an entirely new example.

Reply:
I agree, only reason I went for `Deprecated` was because its FQCN was concise and I thought being a widely used annotation, it would have delivered what we were trying to convey easily.